ot/gromov/_utils.py

```diff
@@ -253,46 +253,78 @@ def gwggrad(constC, hC1, hC2, T, nx=None):
                               T, nx)  # [12] Prop. 2 misses a 2 factor
 
 
-def update_square_loss(p, lambdas, T, Cs):
+def update_square_loss(p, lambdas, T, Cs, nx=None):
     r"""
-    Updates :math:`\mathbf{C}` according to the L2 Loss kernel with the `S` :math:`\mathbf{T}_s`
-    couplings calculated at each iteration
+    Updates :math:`\mathbf{C}` according to the L2 Loss kernel with the `S`
+    :math:`\mathbf{T}_s` couplings calculated at each iteration of the GW
+    barycenter problem in :ref:`[12]`:
+
+    .. math::
+
+        \mathbf{C}^* = \mathop{\arg \min}_{\mathbf{C}\in \mathbb{R}^{N \times N}} \quad \sum_s \lambda_s \mathrm{GW}(\mathbf{C}, \mathbf{C}_s, \mathbf{p}, \mathbf{p}_s)
+
+    Where :
+
+    - :math:`\mathbf{C}_s`: metric cost matrix
+    - :math:`\mathbf{p}_s`: distribution
 
     Parameters
     ----------
     p : array-like, shape (N,)
         Masses in the targeted barycenter.
     lambdas : list of float
         List of the `S` spaces' weights.
-    T : list of S array-like of shape (ns,N)
+    T : list of S array-like of shape (N, ns)
         The `S` :math:`\mathbf{T}_s` couplings calculated at each iteration.
     Cs : list of S array-like, shape(ns,ns)
         Metric cost matrices.
+    nx : backend, optional
+        If let to its default value None, a backend test will be conducted.
 
     Returns
     ----------
     C : array-like, shape (`nt`, `nt`)
         Updated :math:`\mathbf{C}` matrix.
+
+    References
+    ----------
+    .. [12] Gabriel Peyré, Marco Cuturi, and Justin Solomon,
+        "Gromov-Wasserstein averaging of kernel and distance matrices."
+        International Conference on Machine Learning (ICML). 2016.
+
     """
-    T = list_to_array(*T)
-    Cs = list_to_array(*Cs)
-    p = list_to_array(p)
-    nx = get_backend(p, *T, *Cs)
+    if nx is None:
+        T = list_to_array(*T)
```

Comment:
we shoudl remove the list_to_array that is source of bugs and is not necssary for a function that is used inside th solver